Makefile

```diff
@@ -157,6 +169,21 @@ docs: docs/_build/html/index.html
 docs/_build/html/index.html: check test
 	$(MAKE) -C docs/ html
 
+# Prune the packages currently installed in the virtual environment down to the required
+# packages only. Pruning works in a roundabout way, where we first generate the wheels for
+# all installed packages into the build/wheelhouse/ folder. Next we wipe all packages and
+# then reinstall them from the wheels while disabling the PyPI index server. Thus we ensure
+# that the same package versions are reinstalled.
+.PHONY: prune
+prune:
+	mkdir -p build/
```

Comment:
@behnazh should we use [`mktemp -d`](https://linux.die.net/man/1/mktemp) here instead?

Reply:
> @behnazh should we use [`mktemp -d`](https://linux.die.net/man/1/mktemp) here instead?

Don't we want to create a directory called `build`? Using `[mktemp](https://linux.die.net/man/1/mktemp) would use an arbitrary name for the directory, no?

Reply:
I think both approaches would work.

`flit` builds a wheel inside of a temp folder ([src](https://github.com/pypa/flit/blob/f86127764bddb05cb6b5d9a1b6ae4346f786c94a/flit_core/flit_core/wheel.py#L219)), and we create the `build/` folder and then delete it at the end of the recipe so we may as well use a temp folder too? Doesn’t really matter…

Reply:
Let’s keep `build/` for now.

Creating a temp folder, e.g.
```bash
TMP = mktemp -d
```
would require to run all these commands in a single line and in a single shell to ensure that `TMP` is available:
```Makefile
prune:
	TMP = mktemp -d; \
	python ... ; \
```
with the `; \` at the end of each line. Might as well keep it a tad more readable as-is?

Reply:
SGTM.